src/app/exam/ExamInPageNavigationBar.js

```diff
@@ -3,15 +3,16 @@ import {AiOutlineSetting, FaClipboardList, FaRegEdit, FaUserFriends} from "react
 import {useExamContext} from "./problems/ExamContext";
 
 const ExamInPageNavigationBar = function ({currentURL, examName, examId}) {
-    const { setCurrentExam: firstParamResetState } = useExamContext()
+    const {setCurrentExam, setShouldRedirect} = useExamContext()
+    const resetFunctions = [setCurrentExam, setShouldRedirect]
 
     return (
         <InPageNavigationBar currentURL={currentURL}
                              path={{
                                  head: "Exam",
                                  tail: examName
                              }}
-                             events={{ firstParamResetState }}
+                             events={{ firstParamOnClicks: resetFunctions }}
```

Comment:
1. why not put all events as properties (props) of `InPageNavigationBar`

2. `firstParamOnClicks` .. has any other better names?  `onCrumbClick`
where `index` is passed to `onCrumbClick` as `onCrumbClick(index)`

Reply:
This is indeed a nice catch, thank u! :)